### PR TITLE
feat: use tooling packages in isomer components package

### PR DESCRIPTION
**Pull request by karrui**
### TL;DR

This PR uses the eslint and prettier configs introduced in #279 in isomer-components package.  This is the chain of PRs to consolidate tooling to improve DX.

### CODE REVIEW

**Comment by karrui** [comment]
> [!WARNING]
> <b>This pull request is not mergeable via GitHub because a downstack PR is open. Once all requirements are satisfied, merge this PR as a stack <a href="https://app.graphite.dev/github/pr/opengovsg/isomer/280?utm_source=stack-comment-downstack-mergeability-warning" >on Graphite</a>.</b>
> <a href="https://graphite.dev/docs/merge-pull-requests">Learn more</a>

* **#280** <a href="https://app.graphite.dev/github/pr/opengovsg/isomer/280?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* **#279** <a href="https://app.graphite.dev/github/pr/opengovsg/isomer/279?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* `main`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @karrui and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

**Comment by seaerchin** [comment]
136 files... i cry

File: packages/components/eslint.config.mjs
Comment (seaerchin) [inline packages/components/eslint.config.mjs:1]: why use `.mjs` here?

File: packages/components/src/interfaces/internal/LocalSearchInputBox.ts
Comment (seaerchin) [inline packages/components/src/interfaces/internal/LocalSearchInputBox.ts:1]: why prefer `type` vs `interface`? afaik, they are roughly equal?

File: packages/components/src/interfaces/internal/Search.ts [hunk 1]
```diff
@@ -1,4 +1,4 @@
-export type SearchRecord = {
+export interface SearchRecord {
```
Comment (seaerchin) [inline packages/components/src/interfaces/internal/Search.ts:1]: but we swapped from `type` to `interface` here?

File: tooling/eslint/nextjs.js [hunk 2]
```diff
@@ -1,18 +1,16 @@
+import { fixupPluginRules } from "@eslint/compat";
 import nextPlugin from "@next/eslint-plugin-next";
 
 /** @type {Awaited<import('typescript-eslint').Config>} */
 export default [
   {
     files: ["**/*.ts", "**/*.tsx"],
     plugins: {
-      "@next/next": nextPlugin,
+      "@next/next": fixupPluginRules(nextPlugin),
```
Comment (seaerchin) [inline tooling/eslint/nextjs.js:9]: :monkas:

**Review by seaerchin — APPROVED** [review]
lgtm, i think most changes here are from lint

File: packages/components/eslint.config.mjs
Comment (karrui) [reply packages/components/eslint.config.mjs:1]: cause this app is a cjs app for some reason, not fully compatible with esm yet. but i want to use `import` (esm syntax) in eslint so declare as esm lo

File: packages/components/src/interfaces/internal/LocalSearchInputBox.ts
Comment (karrui) [reply packages/components/src/interfaces/internal/LocalSearchInputBox.ts:1]: interface can extend and override, type cannot. but nevertheless, recommended lint just follow HAHA

**Comment by karrui** [comment]
### Merge activity

* **Jul 9, 6:24 AM EDT**: @karrui started a stack merge that includes this pull request via [Graphite](https://app.graphite.dev/github/pr/opengovsg/isomer/280).

**Pull request merged**